internal/exec/engine.go

```diff
@@ -84,17 +85,17 @@ func (e Engine) Run(stageName string) bool {
 		e.logReport(r)
 		if err != nil && err != providers.ErrNoProvider {
 			e.Logger.Crit("failed to acquire default config: %v", err)
-			return false
+			return false, err
 		}
 	default:
 		e.Logger.Crit("failed to acquire config: %v", err)
-		return false
+		return false, err
 	}
 
 	e.Logger.PushPrefix(stageName)
 	defer e.Logger.PopPrefix()
 
-	return stages.Get(stageName).Create(e.Logger, e.Root, *e.Fetcher).Run(config.Append(baseConfig, config.Append(systemBaseConfig, cfg)))
+	return stages.Get(stageName).Create(e.Logger, e.Root, *e.Fetcher).Run(config.Append(baseConfig, config.Append(systemBaseConfig, cfg))), err
```

Comment:
Have you tested this change in a situation where you'd receive a `ErrNoProvider`?

Reply:
In `internal/providers/system/system.go` (line 55) when a config is not given at the default path as specified by `$SYSTEM_CONFIG_DIR`, ErrNoProvider will be called for example, and will show: `Ignition error: config provider was not online`